authmanager/authmanager.go

```diff
@@ -19,21 +20,31 @@ const (
 	pivotalAPITokenMsg = "Please go to https://www.pivotaltracker.com/profile, create a new API token & paste it here:\nToken: "
 	redmineAPITokenMsg = "Please go to %s/my/account, create a new API token & paste it here:\nToken: "
 
+	githubAPITokenWarning = "WARNING: Github has API rate limits for all requests which do not contain a token.\nPlease create a read-only access token to increase that limit. https://github.com/settings/tokens"
+
 	authTokenEnvVariable = "TODOCHECK_AUTH_TOKEN"
 )
 
 // AcquireToken stores the issue tracker's auth token based on the auth type specified
 func AcquireToken(cfg *config.Local) error {
-	switch cfg.Auth.Type {
-	case config.AuthTypeNone:
-		return nil
-	case config.AuthTypeAPIToken:
-		return acquireAPIToken(cfg)
-	case config.AuthTypeOffline:
-		return acquireOfflineToken(cfg.Auth)
-	default:
-		panic("unknown auth token type")
+	err := func() error {
+		switch cfg.Auth.Type {
+		case config.AuthTypeNone:
+			return nil
+		case config.AuthTypeAPIToken:
+			return acquireAPIToken(cfg)
+		case config.AuthTypeOffline:
+			return acquireOfflineToken(cfg.Auth)
+		default:
+			panic("unknown auth token type")
+		}
+	}()
+
+	if cfg.Auth.Token == "" {
+		showAuthWarning(cfg)
```

Comment:
This code should be in [config.Validate](https://github.com/preslavmihaylov/todocheck/blob/ce4a2420e46ddfc5795358363e05399d874a9b4f/config/config.go#L145) because that's the place where we do validations of the user's configuration.

Apart from that, there is no need to change the rest of the `AcquireToken` code, more specifically, the error handling. I don't see how wrapping this in a function helps.

Reply:
Good idea. That's where I started but figured it might be better to do it after we acquire the user's token (which also stores it if the user supplies a token in the CLI). Now that I think about it, it makes more sense to do it up front during validation.